### ISSUE TITLE
ER_DUP_ENTRY thrown in production

### ISSUE THREAD

**Issue by michaelmcmillan** [michaelmcmillan/LittList#41]
``` javascript
{"level":"error","message":"ER_DUP_ENTRY: Duplicate entry 'store norske leksikon' for key 'search'","timestamp":"2015-04-26T11:49:22.385Z"}
{"duration":"3428ms","level":"info","message":"Bibsys query","timestamp":"2015-04-26T11:49:34.372Z"}
{"duration":"2243ms","level":"info","message":"Bibsys query","timestamp":"2015-04-26T11:50:04.672Z"}
```

I suspect that this error is caused by refreshing before the Bibsys query is working.

**Comment by michaelmcmillan** [comment]
ER_DUP_ENTRY: Duplicate entry 'king' for key 'search'

**Comment by michaelmcmillan** [comment]
{"level":"error","message":"ER_DUP_ENTRY: Duplicate entry 'book' for key 'search'","timestamp":"2015-04-27T08:25:07.653Z"}

**Comment by michaelmcmillan** [comment]
This is caused by searching for a non-cached query rapidly. A way to fix this is to have an internal hashmap over queries in the funnel controller in memory that gets cleared every so often (to avoid it becoming monstrous) with setTimeout.

**Comment by michaelmcmillan** [comment]
Even though the error is suppressed queries to external components will still go through.

**Comment by michaelmcmillan** [comment]
This should be fixed after the refactoring of #61.
You need to have one check in the ResultController and one in every component controller.
